walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/openid4vci/OpenId4VciIssuanceRequest.kt

```diff
@@ -23,19 +43,128 @@ internal class OpenId4VciIssuanceRequest(
+                message ?: "User Canceled",
+                VerifiedIdExceptions.USER_CANCELED_EXCEPTION.value
+            )
+        }
+    }
+
+    private suspend fun formatAndSendIssuanceRequest(): RawOpenID4VCIResponse {
+        val accessToken =
+            (requirement as? AccessTokenRequirement)?.accessToken
+                ?: throw OpenId4VciValidationException(
+                    "Access token is missing in requirement.",
+                    VerifiedIdExceptions.REQUEST_CREATION_EXCEPTION.value
+                )
+
+        val credentialEndpoint = credentialMetadata.credential_endpoint
+            ?: throw OpenId4VciValidationException(
+                "Credential endpoint is missing in credential metadata.",
+                VerifiedIdExceptions.MALFORMED_CREDENTIAL_METADATA_EXCEPTION.value
+            )
+
+        val rawRequest = requestFormatter.format(credentialOffer, credentialEndpoint, accessToken)
+        return sendIssuanceRequest(credentialEndpoint, rawRequest, accessToken)
+    }
+
+    private suspend fun sendIssuanceRequest(
+        credentialEndpoint: String,
+        rawRequest: RawOpenID4VCIRequest,
+        accessToken: String
+    ): RawOpenID4VCIResponse {
+        PostOpenID4VCINetworkOperation(
+            credentialEndpoint,
+            libraryConfiguration.serializer.encodeToString(
+                RawOpenID4VCIRequest.serializer(),
+                rawRequest
+            ),
+            accessToken,
+            libraryConfiguration.httpAgentApiProvider,
+            libraryConfiguration.serializer
+        ).fire()
+            .onSuccess { response -> return response }
+            .onFailure {
+                throw OpenId4VciRequestException(
+                    "Failed to send issuance request. ${it.message}",
+                    VerifiedIdExceptions.REQUEST_SEND_EXCEPTION.value
+                )
+            }
+        throw OpenId4VciRequestException(
+            "Failed to send issuance request.",
+            VerifiedIdExceptions.UNSPECIFIED_EXCEPTION.value
+        )
+    }
+
+    private suspend fun mapToVerifiedId(rawResponse: RawOpenID4VCIResponse): VerifiedId {
+        val issuerName =
+            credentialMetadata.transformLocalizedIssuerDisplayDefinitionToRequesterStyle().name
+        val credential = rawResponse.credential ?: throw OpenId4VciValidationException(
+            "Credential is missing in response.",
+            VerifiedIdExceptions.INVALID_PROPERTY_EXCEPTION.value
+        )
+        val raw = verifyAndUnWrapIssuanceResponse(credential)
+        return OpenId4VciVerifiedId(
+            raw = raw,
+            issuerName = issuerName,
+            credentialConfiguration = credentialConfiguration
+        )
+    }
+
+    private suspend fun verifyAndUnWrapIssuanceResponse(jwsTokenString: String): VerifiableCredential {
+        val jwsToken = JwsToken.deserialize(jwsTokenString)
+        if (!verifySignature(jwsToken))
```

Comment:
do we need to verify signature here?

Reply:
It is the response jwt with the VC. I think signature should be verified to check the integrity before unwrapping and saving the VC.